flask_monitoringdashboard/frontend/js/controllers/OverviewController.js

```diff
@@ -28,23 +30,57 @@ export function OverviewController($scope, $http, $location, menuService, endpoi
         $scope.table = response.data;
     });
 
-    function getItemsForPage(pageNumber) {
-        const start = Number(pageNumber) * $scope.pageSize;
-        const end = (Number(pageNumber) + 1) * Number($scope.pageSize);
+    function sortOrder(a, b){
+      return a[$scope.sortBy] > b[$scope.sortBy] || b[$scope.sortBy] === null;
+    }
+    
+    function sortItems(items){
+      return $scope.getOrAddSortingOrder($scope.sortBy) ? items.sort(sortOrder) : items.sort((a, b) => sortOrder(b, a))
+    }
 
+    function getItemsForPage(pageNumber) {
+        const start = pageNumber * Number($scope.pageSize);
+        const end = (pageNumber + 1) * Number($scope.pageSize);
+  
         let items = $scope.table
-            .filter(item => item.name.includes($scope.searchQuery))
+            .filter(item => item.name.includes($scope.searchQuery));
+
         if ($scope.slectedBlueprint) {
-            items = items.filter(item => item.blueprint===$scope.slectedBlueprint)
+            items = items.filter(item => item.blueprint===$scope.slectedBlueprint);
         }
+    
+        return sortItems(items).slice(start, end);
+    }
 
-        return items.slice(start, end);
+    $scope.getOrAddSortingOrder = function (column) {
+        let orderBy = $scope.sortingOrder[column];
+        if (orderBy !== undefined) {
+            return orderBy;
+        }
+        $scope.sortingOrder[column] = true;
+        return true;
+    }
+    
+    $scope.changeSortingOrder = function (column) {
+        for (const [key, value] of Object.entries($scope.sortingOrder))
+            $scope.sortingOrder[key] = key === column ? !value : true;
+        $scope.sortBy = column;
     }
 
     $scope.getFilteredItems = function () {
         return getItemsForPage($scope.currentPage);
     }
 
+    $scope.getSortArrowClassName = function (column) {
+      let isCurrentProp = $scope.sortBy === column;
```

Comment:
could this be named`isCurrentSortingCriterion`? I'm not sure what `prop` means in this context.